ouroboros-network/src/Ouroboros/Network/BlockFetch/ClientRegistry.hs

```diff
@@ -298,7 +298,7 @@ setFetchClientContext :: MonadSTM m
 setFetchClientContext (FetchClientRegistry ctxVar _ _ _ _ _) tracer mkPolicy =
     atomically $ do
       ok <- tryPutTMVar ctxVar (tracer, mkPolicy)
-      unless ok $ error "setFetchClientContext: called more than once"
+      unless ok $ error "setFetchClientContext: called more than once" -- TODO
```

Comment:
This function could afford to be documented with the contract implicit in this error - namely, that it be called exactly (or perhaps at most) once. Conversely, if this contract is incorrect, then perhaps this error could be eliminated.

Reply:
It is only called by `blockFetchLogic`; it is defacto an internal function of the `block-fetch` application.